rust_src/src/chartable.rs

```diff
@@ -156,6 +161,13 @@ impl LispCharTableRef {
     pub fn set_unchecked(self, idx: isize, value: LispObject) {
         unsafe { CHAR_TABLE_SET(self.into(), idx as i32, value) };
     }
+
+    pub fn set_extras(&mut self, idx: usize, value: LispObject) {
```

Comment:
It would make sense to replace `set_char_table_extras` with a wrapper around this method.

Reply:
Better I think to port the C code over first. No sense in C -> Rust -> C here.